File: 0078-subsets/0078-subsets.js
/**
 * @param {number[]} nums
 * @return {number[][]}
 */
var subsets = function(nums) {
    const res = [];
    const loop = (arr, i) => {
        res.push(arr);
        for (let j = i; j < nums.length; j++) {
            arr.push(nums[j]);
            loop([...arr], j + 1);
            arr.pop();
        }
    }
    loop([], 0);
    return res;
};

File: 0078-subsets/0078-subsets.py
class Solution:
    def subsets(self, nums: List[int]) -> List[List[int]]:
        res = []
        def loop(ar, i):
            res.append(ar)
            for j in range(i, len(nums)):
                ar.append(nums[j])
                loop(copy.deepcopy(ar), j + 1)
                ar.pop()
        loop([], 0)
        return res